server/services/event.js

```diff
@@ -299,16 +299,31 @@ module.exports = class EventService {
             playerIdAttackers: attackers.map(p => p._id),
             starId: star._id,
             starName: star.name,
-            combatResult,
             captureResult
         };
 
         for (let defender of defenders) {
-            await this.createPlayerEvent(gameId, gameTick, defender._id, this.EVENT_TYPES.PLAYER_COMBAT_STAR, data);
+            let defenderCombatResult = combatResult
+            defenderCombatResult.carriers = combatResult.carriers.map(c => {
+                if (c.specialist && c.specialist.name === 'Scrambler' && !defender._id.equals(c.ownedByPlayerId) && c.after !== 0) {
+                    return { ...c, before: '???', after: '???' }
```

Comment:
Damage is distributed to carriers with the largest number of ships first, we should mask the `lost` property too as its possible for players to figure out if carriers have more or less ships than another carrier when multiple carriers are involved in combat.

See `combat.js` line 171 for proof of sorting.

Reply:
What about randomize it?

Reply:
As in they all lose at random, and all have a random chance to take the extra damage